### PR TITLE
HOTFIX: Fixes to javadoc and to state store name for link joins

### PR DESCRIPTION
**Pull request by enothereska**

### CODE REVIEW

**Comment by enothereska** [comment]
@guozhangwang @dguy have a look when you can please. Thanks.

File: streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java [hunk 1]
```diff
@@ -425,7 +425,7 @@
     /**
      * Combine values of this stream with another {@link KStream}'s elements of the same key using windowed Outer Join.
      * If a record key is null it will not included in the resulting {@link KStream}
-     * Both of the joining {@link KStream}s will be materialized in local state stores with the given store names.
+     * Both of the joining {@link KStream}s will be materialized in local state stores with with auto-generated name ${storeName}.
```
Comment (dguy) [inline streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java]: typo: ..with with..

same typo also in the other javadoc comments below.
Also, where does ${storeName} come from? Would it be enough just to say "with an auto-generated store name" ?

**Comment by enothereska** [comment]
Thanks @dguy

**Comment by dguy** [comment]
LGTM

**Comment by guozhangwang** [comment]
LGTM. Merged to trunk, thanks @enothereska !

**Pull request closed without merging**